### PR TITLE
Bump gmp images to v0.5.0

### PR DESCRIPTION
**Pull request by realschwa**
Changing the image versions to correspond with our upcoming release

### CODE REVIEW

File: cmd/operator/deploy/operator/05-deployment.yaml [hunk 1]
```diff
@@ -41,7 +41,7 @@ spec:
       automountServiceAccountToken: true
       containers:
       - name: operator
-        image: gke.gcr.io/prometheus-engine/operator:v0.4.3-gke.0
+        image: gke.gcr.io/prometheus-engine/operator:v0.5.0
```
Comment (pintohutch) [inline cmd/operator/deploy/operator/05-deployment.yaml:44]: these need the `-gke.0` suffix

Comment (realschwa) [reply cmd/operator/deploy/operator/05-deployment.yaml:44]: will add back

File: cmd/operator/deploy/operator/10-collector.yaml [hunk 2]
```diff
@@ -108,7 +108,7 @@ spec:
           seccompProfile:
             type: RuntimeDefault
       - name: config-reloader
-        image: gke.gcr.io/prometheus-engine/config-reloader:v0.4.1-gke.0
+        image: gke.gcr.io/prometheus-engine/config-reloader:vv0.5.0-gke.0
```
Comment (pintohutch) [inline cmd/operator/deploy/operator/10-collector.yaml:111]: typos `vv` throughout

Comment (realschwa) [reply cmd/operator/deploy/operator/10-collector.yaml]: whoooopsie

**Review by pintohutch — APPROVED** [review]

**Pull request merged**